src/fat/fs.rs

```diff
@@ -380,14 +389,15 @@ impl Default for FileFilter {
     }
 }
 
-type SyncSectorBufferFn<S> = fn(&FileSystem<S>) -> Result<(), <S as ErrorType>::Error>;
-type UnmountFn<S> = fn(&FileSystem<S>) -> FSResult<(), <S as ErrorType>::Error>;
+type SyncSectorBufferFn<S, C> = fn(&FileSystem<S, C>) -> Result<(), <S as ErrorType>::Error>;
+type UnmountFn<S, C> = fn(&FileSystem<S, C>) -> FSResult<(), <S as ErrorType>::Error>;
 
 /// An API to process a FAT filesystem
 #[derive(Debug)]
-pub struct FileSystem<S>
+pub struct FileSystem<S, C>
 where
     S: Read + Seek,
+    C: Clock,
```

Comment:
I can see that this was done for reducing unnecessary allocations, but then again, I am afraid this won't behave well if a custom clock is to be provided

Reply:
then just not pick the commit

Reply:
Yes, in that case a FSOptions::new(clock) would work. But I leave it to you. Like I said, I did this to reduce allocation and even though I (at least currently) don't intend to use or further add to this project I'd wanted to share it.